src/ProductView/TabBar/QueryInterface/QueryInterface.js

```diff
@@ -118,10 +122,18 @@ class QueryInterface extends Component {
     }
 
     handleChangeQuery(event) {
-        this.setState({ eventQuery: event.target.value });
+        const queryValue = event.target.value;
+        const modalIsAbleToSave = this.isModalAbleToSave(this.state.eventTypeName, queryValue);
+        this.setState({ eventQuery: queryValue, modalIsAbleToSave: modalIsAbleToSave });
+        this.eventQueryFormValidation(queryValue);
```

Comment:
Does nearly the same stuff like isModalAbleToSave(). Plese consider building one function for both operations.